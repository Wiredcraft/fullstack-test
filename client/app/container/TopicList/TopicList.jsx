import React, { Component } from 'react';
import { connect } from 'react-redux';
import {bindActionCreators} from 'redux';
import { Loader, Topic } from '../../components/index';
import * as TopicsActions from '../../actions/TopicsActions';

import classNames from 'classnames';
import styles from '../../scss/app.scss';
import stylesPriv from './_TopicList.scss';

@connect(
  state => ({ topics: state.topics, auth: state.auth }),
  dispatch => bindActionCreators(TopicsActions, dispatch)
)
export default class TopicList extends Component {

  _Loader(){
      return (this.props.topics.loading) ? <div className={styles['center']}><Loader /></div> : '';
  }

  _isEmpty(){
    if(this.props.topics.topics.length === 0){
      return (
        <div className={classNames(styles['h2'], styles['center'])}>
            List is empty
            <div className={styles['center']}>
              <button className={stylesPriv['topic-list__btn']}>POST NOW</button>
            </div>
        </div>
      )
    }
  }

  _isOwned(id){
    return (this.props.auth.id === id);
  }

  // open close
  // like

  _List(){
      let that = this;
      return _.map(_.sortByOrder(this.props.topics.topics, ['points'], ['desc']), (item, key) => {
        return <Topic key={key} {...item} num={key + 1} owned={that._isOwned(item.authorId)} />
      });
  }

  render() {

    return (
      <div>
        {this._Loader()}
        {this._isEmpty()}
        {this._List()}
      </div>
    );
  }
}
