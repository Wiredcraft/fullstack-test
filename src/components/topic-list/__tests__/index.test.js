import React from 'react';
import renderer from 'react-test-renderer';

import TopicList from '../index';

test('TopicList renders', async () => {
  const component = renderer.create(<TopicList />);

  expect(component.toJSON()).toMatchSnapshot();
});
