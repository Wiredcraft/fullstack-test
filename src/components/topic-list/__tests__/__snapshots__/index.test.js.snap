// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicList renders 1`] = `
<table
  className="sc-pVTFL jKYrNm"
>
  <tbody />
</table>
`;
